### PR TITLE
clickhouse: add article on sharding

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/DOC-42

The main purpose of this article is showing the user HOW TO use shards with Aiven for ClickHouse. Since there is no conceptual article on sharing  to link to (and perhaps no need to create a dedicated separate one on this) and I still feel that the user need some basic info on the subject before going into actual actions, I'm also adding some short background on what is sharding, why use it, and how it works with Aiven. It's followed by the instruction on how to use the shards.

### CODE REVIEW

**Review by lornajane — CHANGES_REQUESTED** [review]
Can we please review the content type on this document? It has some conceptual information and then more of a task-based section. The `CONTRIBUTING` guide has more detail on the content framework we use and how to structure information for users, and I'm also happy to answer any questions you have on this.

**Comment by wojcik-dorota** [comment]
> 

@lornajane , this is a how-to article. As you suggested, I cut down on the overview section - now there are three introductory sentences, which are followed by the steps.

**Comment by aris-tritas-aiven** [comment]
Hi, please resolve the warnings in https://app.netlify.com/sites/developer-aiven-io-preview/deploys/636ce95fb1b9550008c2fac9#L1354

**Comment by wojcik-dorota** [comment]
> Hi, please resolve the warnings in [https://app.netlify.com/sites/developer-aiven-io-](https://app.netlify.com/sites/developer-aiven-io-preview/deploys/636ce95fb1b9550008c2fac9#L1354)

@aris-tritas-aiven , fixed all that I could find. Thank you!

**Review by anelook — CHANGES_REQUESTED** [review]
This PR contains two different set of changes. To avoid future confusion, ease any reverting and keep the clear history of changes, it should be separated and the title should me updated to describe the changes.

**Comment by wojcik-dorota** [comment]
> This PR contains two different set of changes. To avoid future confusion, ease any reverting and keep the clear history of changes, it should be separated and the title should me updated to describe the changes.

@anelook , thank you - moved the fixes to https://github.com/aiven/devportal/pull/1554

**Comment by aris-tritas-aiven** [comment]
When the link fixing PR goes in, you will need to rebase this off the `main` branch.

**Comment by wojcik-dorota** [comment]
> 

@aris-tritas-aiven it's done! Thank you :-)

File: docs/products/clickhouse/howto/sharding.rst
Comment (anelook) [inline docs/products/clickhouse/howto/sharding.rst:30]: We haven't used "topic" anywhere else.  What is information goal of this section? It doesn't seem to fit into "howto", are we talking about concept?

File: docs/products/clickhouse/howto/sharding.rst
Comment (anelook) [inline docs/products/clickhouse/howto/sharding.rst:25]: How do we "Create cluster *test-cluster* with 5 shards"? I'm not sure it is clear for me how to fulfil this prerequisite.

File: docs/products/clickhouse/howto/sharding.rst
Comment (anelook) [inline docs/products/clickhouse/howto/sharding.rst:50]: This might be picky - but the whole CREATE TABLE is table structure, not the list of columns. Also, I think general link to tutorial is not the best reference to point to available types and properties. https://clickhouse.com/docs/en/sql-reference/data-types/ or https://clickhouse.com/docs/en/sql-reference will be a better fit in my opinion

File: docs/products/clickhouse/howto/sharding.rst
Comment (anelook) [inline docs/products/clickhouse/howto/sharding.rst:1]: Is it in ClickHouse or Aiven for ClickHouse? These are very different things and should be taken with care to avoid legal issues.

**Review by anelook — CHANGES_REQUESTED** [review]
Thank you for adding this article. Information on how to perform actions with sharding with Aiven for ClickHouse is something we indeed need for the GA!

I have couple of questions. The biggest challenge for me when reading this article was to grasp what it is actually about - 
Is it about general sharding or Aiven for ClickHouse? 
What question this article intends to answer? 

Many of our users will come from google, what would be the "how to .." problem they are looking for when they land here? And how we can help answering their question in a simple, short but concise way. I feel there are a lot of extra bits of knowlege, but it made it very difficult to get to the "meat" of the article... 

How to work with shards in ClickHouse is very vague, and therefore will not be very helpful for the developers who just want to solve a specific problem (also not good for SEO, which works better for focused content). Can we narrow it?

File: docs/products/clickhouse/howto/use-shards-with-distributed-table.rst
Comment (anelook) [inline docs/products/clickhouse/howto/use-shards-with-distributed-table.rst:36]: This label is no longer used I believe

File: docs/products/clickhouse/howto/use-shards-with-distributed-table.rst
Comment (anelook) [inline docs/products/clickhouse/howto/use-shards-with-distributed-table.rst:50]: There is something wrong with consistency "-" vs "\_" in the names given to table and to db. Seems to me like a typo.
Generally whenever possible for Clickhosue better to use "\_", because "-" won't work with ClickHouse and needs to be surrounded in back quotes "\`" \`like-this\`.

**Review by anelook — CHANGES_REQUESTED** [review]
@wojcik-dorota Thank you so much for adding the changes! The article is easy to follow, I enjoyed reading it and following the steps. 
I left couple of suggestions and minor corrections.

One more thing - the title of PR, the "MVP part" do we need it for clean history? I think it fits better to description including [DOC-42].

File: docs/products/clickhouse/howto/use-shards-with-distributed-table.rst
Comment (wojcik-dorota) [reply docs/products/clickhouse/howto/use-shards-with-distributed-table.rst:4]: yeah, definitely, the concept for sharding would need some diagram showing cluster - nodes - shards interdependencies.

**Review by kmichel-aiven — DISMISSED** [review]

File: docs/products/clickhouse/howto/use-shards-with-distributed-table.rst
Comment (wojcik-dorota) [reply docs/products/clickhouse/howto/use-shards-with-distributed-table.rst:13]: @kmichel-aiven , it's enough to use a product name with ® once in an article to fulfill legal requirements. Since I've already used it in the article, I don't need to use it here.

File: docs/products/clickhouse/howto/use-shards-with-distributed-table.rst
Comment (wojcik-dorota) [reply docs/products/clickhouse/howto/use-shards-with-distributed-table.rst:39]: @kmichel-aiven , it's enough to use a product name with ® once in an article to fulfill legal requirements. Since I've already used it in the article, I don't need to use it here.

**Review by anelook — APPROVED** [review]
This looks good to me! Thank you @wojcik-dorota for adding all the improvements. Also I noticed that @kmichel-aiven had a look as well, makes me happy, to get any notes on potential technical incorrectness :) 
From my side it can be merged, so I approve. I see other reviewers in the list, so won't merge in case some reviews are still pending.

**Comment by lornajane** [comment]
@wojcik-dorota is there anyone else you'd like to have approval from before we merge?

**Comment by lornajane** [comment]
Sorry, I am not merging tonight as I noticed some warnings regarding the formatting. Please check the tiny tweaks and if you're happy, let's get this live :)

**Comment by wojcik-dorota** [comment]
> Sorry, I am not merging tonight as I noticed some warnings regarding the formatting. Please check the tiny tweaks and if you're happy, let's get this live :)

@lornajane - thanks for the fixes! I can see no more warnings so we're good to go now :-)

**Comment by kmichel-aiven** [comment]
We should wait a bit before merging, the feature is not available yet.

(I should have realized that earlier :facepalm:)

**Comment by lornajane** [comment]
It's a fairly key requirement that users should be able to do whatever it is we're documenting!! I'll switch this to draft for now and we can revisit at the appropriate time.

**Comment by kmichel-aiven** [comment]
This does not need to be a draft anymore, sharding is available :rocket:

**Pull request merged**
